generator/src/swam/generator/Generator.scala

```diff
@@ -0,0 +1,90 @@
+package swam.generator
+
+import java.io.File
+import java.util.concurrent.Executors
+
+import cats.effect.{Blocker, IO}
+import org.json4s.DefaultFormats
+import swam.runtime.Engine
+import org.json4s.jackson.Serialization.writePretty
+
+import scala.concurrent.ExecutionContext
+case class Config(wasms: Seq[File] = Seq(),
+                  printTemplateContext: Boolean = false,
+                  createBoilerplate: String = "",
+                  renderTemplate: File = null)
+
+/**
+    @author Javier Cabrera-Arteaga on 2020-03-07
+  */
+object Generator extends App {
```

Comment:
You should use [`IOApp`](https://typelevel.org/cats-effect/datatypes/ioapp.html), which is intended exactly to have an `IO` based main, with all the `ContextShift` and friends you want to have.